doc/rtd/howto/debugging.rst

```diff
@@ -0,0 +1,142 @@
+.. _how_to_debug:
+
+How to debug cloud-init
+***********************
+
+There are several cloud-init failure modes that one may need to
+debug. Debugging is specific to the scenario, but the starting points are often
+similar.
+
+I can't log in to my instance
+=============================
+
+One of the more challenging scenarios to debug is when you don't have
+shell access to your instance. You have a few options:
+
+1. Acquire console log messages and check for errors.
+
+2. Try running the same user-data locally, such as in one of the
+   :ref:`tutorials<tutorial_index>`. Use LXD or Qemu locally to get a shell or
+   logs then debug with :ref:`these steps<did_not_do_the_thing>`.
+
+3. Try copying the image to your local system, mount the filesystem locally
+   and inspect (create a chroot in the image filesystem before attempting the
+   steps in other sections).
```

Comment:
> When you say "copy the image", do you mean the one used in the previous step? If they don't follow the tutorial, what image should they use?

Not necessarily. One can copy any image being used to launch the instance. This might mean exporting the image from a cloud to copy it locally, or yes if using the tutorial one could copy the image used in that.